javalib/src/main/scala/java/lang/StrictMath.scala

```diff
@@ -0,0 +1,187 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+/*
+ * ====================================================
+ * Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+ *
+ * Developed at SunSoft, a Sun Microsystems, Inc. business.
+ * Permission to use, copy, modify, and distribute this
+ * software is freely granted, provided that this notice
+ * is preserved.
+ * ====================================================
+ */
+
+package java.lang
+
+object StrictMath {
+
+  // Ported from https://www.netlib.org/fdlibm/s_floor.c
+  def floor(x: scala.Double): scala.Double = {
+    val doubleBits = Double.doubleToLongBits(x)
```

Comment:
```suggestion
    val doubleBits = Double.doubleToRawLongBits(x)
```
No need to normalize NaNs. The raw variant is faster.